tests/performance/lapack_dstedc.cpp

```diff
@@ -0,0 +1,205 @@
+/*
+ *   Copyright 2021 Huawei Technologies Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cmath>
+#include <iostream>
+#include <sstream>
+#include <vector>
+
+#include <graphblas/utils/Timer.hpp>
+
+#include "lapacke.h"
+
+typedef double ScalarType;
+constexpr ScalarType tol = 1.e-10;
+constexpr size_t RNDSEED = 1;
+
+struct inpdata {
+	size_t N=0;
+  	size_t repeat=1;
+};
+
+void print(const char * name, const double* matrix, int N)
+{
+  printf("\nMatrix %s size %d :\n", name, N);
+  printf(" %s = array ( [", name);
+  for (int i = 0; i < N; i++){
+  printf("\n  [");
+    for (int j = 0; j < N; j++){
+      printf("%.10f, ", matrix[j*N + i]);
+    }
+    printf(" ],");
+  }
+  printf("\n])\n");
+}
+
+
+//** generate vector or upper/lower triangular part of an SPD matrix */
+template< typename T >
+void generate_vec_or_spd_matrix_full( size_t N, std::vector<T> &data ) {
+	if( data.size() == N ) {
+		for( size_t i = 0; i < N; ++i ) {
+			data[ i ] = static_cast< T >( std::rand() ) / static_cast< T >( RAND_MAX );
+		}
+	} else if( data.size() == N * N ) {
+		for( size_t i = 0; i < N; ++i ) {
+			for( size_t j = 0; j < N; ++j ) {
+				size_t k = i * N + j;
+				if( i <= j ) {
+					data[ k ] = static_cast< T >( std::rand() ) / static_cast< T >( RAND_MAX );
+				}
+				if( i == j ) {
+					data[ k ] = data[ k ] + static_cast< T >( N );
+				}
+				if( i > j ) {
+					data[ i * N + j ] = data[ j * N + i ];
+				}
+
+			}
+		}
+	} else {
+		std::cout << "Error: generate_spd_matrix_full: Provided container does not have adequate size\n";
+	}
+
+}
+
+
+
+void alp_program( const inpdata &unit, bool &rc ) {
+	rc = true;
+
+	int N = unit.N;
+	grb::utils::Timer timer;
+	timer.reset();
+	double times;
+
+
+	std::cout << "Testing dstedc_  ( " << N << " x " << N << " )\n";
+	std::cout << "Test repeated " << unit.repeat << " times.\n";
+
+	char compz='I';
+	std::vector< ScalarType > vec_d( N );
+	generate_vec_or_spd_matrix_full( N, vec_d );
+	std::vector< ScalarType > vec_e( N - 1 );
+	generate_vec_or_spd_matrix_full( N - 1, vec_e );
+	std::vector< ScalarType > mat_z( N * N );
+	ScalarType wopt;
+	int lwork = -1;
+	int iwopt;
+	int liwork = -1;
+	int info;
+	
+	dstedc_(&compz, &N, &( vec_d[0] ), &( vec_e[0] ), &( mat_z[0] ), &N, &wopt, &lwork, &iwopt, &liwork, &info);
```

Comment:
I'm not sure if `dstedc_` needs to be called twice, to get `lword` and `liwork` separately ?